### PR TITLE
better errors: check for callability of lax.control_flow arguments

### PR DESCRIPTION
**Pull request by jakevdp**
Improves errors for issue raised in #10258

### CODE REVIEW

**Review by mattjj — APPROVED** [review]
LGTM! Could add the word "argument" to the error messages, like "lax.scan: f argument should be a callable function". Whatever you think reads best!

**Comment by mattjj** [comment]
Just realized, we could add error message tests for these. Optional.

File: jax/_src/lax/control_flow.py [hunk 1]
```diff
@@ -193,6 +193,8 @@ def fori_loop(lower, upper, body_fun, init_val):
   Returns:
     Loop value from the final iteration, of type ``a``.
   """
+  if not callable(body_fun):
+    raise TypeError("lax.fori_loop: body_fun and cond_fun should be callable functions.")
```
[the remark this answers is not in the document]
Comment (jakevdp) [reply jax/_src/lax/control_flow.py]: Done, thanks

**Comment by jakevdp** [comment]
> Just realized, we could add error message tests for these

Done

**Pull request merged**